adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractHonoResource.java

```diff
@@ -562,6 +571,7 @@ protected void addCommandToResponse(
                         commandSample);
                 LOG.debug("command message is invalid: {}", command);
                 commandContext.reject("malformed command message");
+                processCommandSpan.finish();
```

Comment:
can't we simply pull this out of the top level *if* statement?

Reply:
Not really - the first of the 3 `processCommandSpan.finish()` invocations (in line 548) is inside an `onComplete` block, so it's invoked some time later.

Reply:
oh, right